frappe/utils/data.py

```diff
@@ -2238,3 +2240,28 @@ def add_trackers_to_url(url: str, source: str, campaign: str, medium: str = "ema
 
 	url_parts[4] = urlencode(query)
 	return urlunparse(url_parts)
+
+
+def decode_sequence(encoded_sequence) -> str:
+	"""
+	Decodes a encoded_sequence consisting of a tuple (string, charset_encoding). The function concatenates all chunks and returns the resulting decoded string.
+
+	Args:
+		encoded_sequence ((string, charset_encoding)): A list of tuples where each tuple contains a chunk of the string and its encoding.
+
+	Returns:
+		str: The decoded and concatenated sequence string.
+	"""
+	from frappe import safe_decode
+	decoded_string = ""
+	for chunk, encoding in encoded_sequence:
+		if encoding is not None:
+			decoded_chunk = safe_decode(param=chunk, encoding=encoding)
+		else:
+			if not isinstance(chunk, str):
+				detected_encoding = chardet.detect(chunk)["encoding"]
+				decoded_chunk = safe_decode(param=chunk, encoding=detected_encoding)
+			else:
+				decoded_chunk = safe_decode(param=chunk)
```

Comment:
this branching is unnecessary? we're trying to decode a string?

in decode_header's definition: https://github.com/python/cpython/blob/main/Lib/email/header.py#L62-L65